src/debugging/DockerDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,91 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken, debug, DebugConfiguration, DebugConfigurationProvider, ProviderResult, WorkspaceFolder } from 'vscode';
+import { callWithTelemetryAndErrorHandling, IActionContext } from 'vscode-azureextensionui';
+import { quickPickWorkspaceFolder } from '../utils/quickPickWorkspaceFolder';
+import { DockerClient } from './coreclr/CliDockerClient';
+import { DebugHelper, DockerDebugContext, ResolvedDebugConfiguration } from './DebugHelper';
+import { DockerPlatform, getPlatform } from './DockerPlatformHelper';
+import { NetCoreDockerDebugConfiguration } from './netcore/NetCoreDebugHelper';
+import { NodeDockerDebugConfiguration } from './node/NodeDebugHelper';
+
+export interface DockerDebugConfiguration extends NetCoreDockerDebugConfiguration, NodeDockerDebugConfiguration {
+    platform?: DockerPlatform;
+}
+
+export class DockerDebugConfigurationProvider implements DebugConfigurationProvider {
+    constructor(
+        private readonly dockerClient: DockerClient,
+        private readonly helpers: { [key in DockerPlatform]: DebugHelper }
+    ) { }
+
+    public provideDebugConfigurations(folder: WorkspaceFolder | undefined, token?: CancellationToken): ProviderResult<DebugConfiguration[]> {
+        return undefined;
+    }
+
+    public resolveDebugConfiguration(folder: WorkspaceFolder | undefined, debugConfiguration: DockerDebugConfiguration, token?: CancellationToken): ProviderResult<DebugConfiguration | undefined> {
+        const debugPlatform = getPlatform(debugConfiguration);
+        return callWithTelemetryAndErrorHandling(
+            `docker-launch/${debugPlatform || 'unknown'}`,
+            async (actionContext: IActionContext) => await this.resolveDebugConfigurationInternal(
+                {
+                    folder: folder || await quickPickWorkspaceFolder('To debug with Docker you must first open a folder or workspace in VS Code.'),
+                    platform: debugPlatform,
+                    actionContext: actionContext
+                },
+                debugConfiguration));
+    }
+
+    private async resolveDebugConfigurationInternal(context: DockerDebugContext, originalConfiguration: DockerDebugConfiguration): Promise<DockerDebugConfiguration | undefined> {
+        context.actionContext.telemetry.properties.platform = context.platform;
+
+        context.folder = context.folder || await quickPickWorkspaceFolder('To debug with Docker you must first open a folder or workspace in VS Code.');

```

Comment:
>quickPickWorkspaceFolder [](start = 49, length = 24)

When does this come up?

Reply:
Never, since we prompt above. I'll remove this.